client/src/components/breadcrumbs.jsx

```diff
@@ -0,0 +1,56 @@
+import { useState, useEffect } from 'react'
+import PropTypes from 'prop-types'
+import './Breadcrumbs.css'
+
+// internal logic for breadcrumbs:
+// "/"   -->  ["home"]
+// "/one" --> ["home", "one"]
+// "/one/one_one" --> ["home", "one", "one_one"] --> home / one / one_one
+//
+// hint:
+// ["home", "one", "one_one"].join("/")
+
+const Breadcrumbs = ({ currentPath, fetchFiles }) => {
+  const [breadcrumbs, setBreadcrumbs] = useState([])
+
+  useEffect(() => {
+    const pathnames = currentPath === '/' ? [``] : currentPath.split('/').slice(1)
+    console.log('Current Path:', currentPath)
+
+    const newBreadcrumbs = [{ name: 'home', path: '/' }]
+
+    pathnames.forEach((name, index) => {
+      if (name) {
+        const path = `/${pathnames.slice(0, index + 1).join('/')}`
+        newBreadcrumbs.push({ name, path })
+      }
+    })
+
+    setBreadcrumbs(newBreadcrumbs)

```

Comment:
is it possible to get rid of state in this component and avoid using it.
if yes, why?

Reply:
how about this implementation? 
```js
  map((folderName, idx, arr) => {
      const folderPath = `/${arr.slice(0, idx + 1).join("/")}`
      return <li onClick={() => fetchFiles(folderPath)}>{folderName}<li>
  })
```

Reply:
Your version is more readable and concise